### PR TITLE
Fix #2117 - Implement LiveRealmData for ScheduleFragmentViewModel.

### LINKED ISSUE #2117: Implement LiveRealmData for ScheduleFragmentViewModel
**Actual Behaviour**

LiveRealmData is not implemented yet.

**Expected Behaviour**

Implement LiveRealmData.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by mayank8318**
Fixes #2117 

Changes: implement LiveRealmData for ScheduleFragmentViewModel

Screenshots for the change:

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java [hunk 1]
```diff
@@ -1,93 +1,67 @@
...
-        if (tracksRealmResults != null) {
-            tracksRealmResults.removeAllChangeListeners();
-        }
-    }
-
-    @Override
-    protected void onCleared() {
-        clearListeners();
-        super.onCleared();
+    private void subscribeToTracks() {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java]: Use RealmDataWrapper

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java (hunk 1 above)
Comment (mayank8318) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java]: @iamareebjamal Sorry didn't get you?

Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java]: There are new helper classes in the project which abstract the functionality of managing Realm connections, please use those

**Comment by mayank8318** [comment]
@iamareebjamal Please review :)

File: android/app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java [hunk 2]
```diff
@@ -91,9 +91,11 @@ protected int getLayoutResource() {
     private void setupViewPager(final ViewPager viewPager) {
         adapter = new ScheduleViewPagerAdapter(getChildFragmentManager());
 
-        scheduleFragmentViewModel.getEventDateString().observe(this, datePair -> {
-            adapter.addFragment(new DayScheduleFragment(), datePair.first, datePair.second);
-            adapter.notifyDataSetChanged();
+        scheduleFragmentViewModel.getEventDateString().observe(this, pairs -> {
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/fragments/ScheduleFragment.java]: Rename pairs to datePair

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java [hunk 3]
```diff
@@ -1,93 +1,60 @@
...
-            eventDatesRealmResults = realmRepo.getEventDates();
-            addEventDatesRealmResultsChangeListener();
-        } else {
-            for (int i = 0; i < eventDateStringData.size(); i++) {
-                eventDateStringLivePair.setValue(eventDateStringData.get(i));
-            }
-        }
+    public LiveData<List<Pair<String, String>>> getEventDateString() {
+        if (eventDateStringLivePair == null)
+            subscribeToEventDateString();
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java]: Just write the method content directly in here

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by mayank8318** [comment]
@iamareebjamal Please review :)

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**